django/db/backends/mysql/operations.py

```diff
@@ -195,21 +195,51 @@ def return_insert_columns(self, fields):
 
     def sql_flush(self, style, tables, sequences, allow_cascade=False):
         # NB: The generated SQL below is specific to MySQL
-        # 'TRUNCATE x;', 'TRUNCATE y;', 'TRUNCATE z;'... style SQL statements
-        # to clear all tables of all data
+        # 'TRUNCATE x;', 'TRUNCATE y;', 'DELETE FROM z;'... style SQL
+        # statements to clear all tables of all data.
+        # 'DELETE FROM' is used for small tables, 'TRUNCATE' is used for
+        # large tables.
```

Comment:
Based on [this comment](https://code.djangoproject.com/ticket/31275#comment:10) I think the same effect can be achieved without the 1000 rows heuristics

```python
def sql_flush(self, style, tables, sequences, allow_cascade=False):
    if not tables:
        return
    sql = ['SET FOREIGN_KEY_CHECKS = 0;']
    tables = set(tables)
    # It's faster to TRUNCATE tables that require a sequence reset since
    # ALTER TABLE AUTO_INCREMENT is slower than TRUNCATE.
    sql.extend(
        '%s %s;' % (
            style.SQL_KEYWORD('TRUNCATE'),
            style.SQL_FIELD(self.quote_name(table_name)),
        ) for table_name in tables.intersection(sequences)
    )
    # Otherwise issue a simple DELETE since it's faster than TRUNCATE
    # and preserves sequences.
    sql.extend(
        '%s %s %s;' % (
            style.SQL_KEYWORD('DELETE'),
            style.SQL_KEYWORD('FROM'),
            style.SQL_FIELD(self.quote_name(table_name)),
        ) for table_name in tables.difference(sequences)
    )
    sql.append('SET FOREIGN_KEY_CHECKS = 1;')
    return sql
```

The `sequence_reset_by_name_sql` addition still needs to hang around even if not used by `sql_flush` since it's necessary for the `TransactionTestCase.reset_sequence` feature to work.

Reply:
Thank you for your comment! I updated this patch.